clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1216,6 +1216,9 @@ event_bb_app2app(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
         DR_ASSERT(false);
         /* in release build, carry on: we'll just miss per-iter refs */
     }
+    if (!drx_expand_scatter_gather(drcontext, bb, NULL)) {
```

Comment:
Xref my comment at https://github.com/DynamoRIO/dynamorio/issues/2985#issuecomment-826920909 -- I think online traces will have the wrong instr sizes until #4865 is in place (another area where we don't have good detailed testing...all we have are vague "did it print some counts" and no checks on types or sizes).  Prob worth a `XXX #4865` comment for online traces.

Reply:
Done.